matrix_sdk_base/src/store/sled_store/mod.rs

```diff
@@ -894,15 +1433,50 @@ impl StateStore for SledStore {
     async fn remove_media_content_for_uri(&self, uri: &MxcUri) -> Result<()> {
         self.remove_media_content_for_uri(uri).await
     }
+
+    async fn get_timeline(
+        &self,
+        room_id: &RoomId,
+        start: Option<&EventId>,
+        end: Option<&EventId>,
+        limit: Option<usize>,
+        direction: Direction,
+    ) -> Result<Option<StoredTimelineSlice>> {
+        self.get_timeline(room_id, start, end, limit, direction).await
+    }
+
+    async fn remove_timeline(&self, room_id: Option<&RoomId>) -> Result<()> {
+        self.remove_timeline(room_id).await
+    }
+}
+
+fn deserialize_event<T: for<'b> Deserialize<'b>>(
```

Comment:
Why is this needed? If it is indeed needed does it need to take an owned version of the store key? Doesn't seem so? Can we modify `SledStore::deserialize_event()` so that it uses this method?

Reply:
hmm, rustc didn't like to move self into a closure. Don't remember what the issue was. I will remove the additional function.